main.go

```diff
@@ -82,3 +118,20 @@ func fetchAllSongs(context *gin.Context) {
 
 	context.JSON(http.StatusOK, gin.H{"status": http.StatusOK, "data": songs})
 }
+
+
+func fetchSong(context *gin.Context) {
+	var song songModel
+	songID := context.Param("id")
+
+	db.First(&song, songID)
+
+	if song.ID == 0 {
```

Comment:
Will this cover edge cases where the ID is something other than 0, such as null/nill?